core/src/test/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizerTest.java

```diff
@@ -394,19 +392,18 @@ void push_page_size_noop_if_no_sub_plans() {
 
   @Test
   void table_scan_builder_support_offset_push_down_can_apply_its_rule() {
-    when(table.createPagedScanBuilder(anyInt())).thenReturn(pagedTableScanBuilder);
+    when(tableScanBuilder.pushDownPageSize(any())).thenReturn(true);
 
     var relation = new LogicalRelation("schema", table);
     var optimized = LogicalPlanOptimizer.create()
         .optimize(new LogicalPaginate(42, List.of(project(relation))));
     // `optimized` structure: LogicalPaginate -> LogicalProject -> TableScanBuilder
     // LogicalRelation replaced by a TableScanBuilder instance
-    assertEquals(project(pagedTableScanBuilder), optimized);
+    assertEquals(project(tableScanBuilder), optimized);
```

Comment:
Please update the comment above: `LogicalPaginate` is removed from the optimized tree